RecoverCRPVM/CreateCRPRescueVM.ps1

```diff
@@ -0,0 +1,260 @@
+﻿<#
+.SYNOPSIS
+    Creates a Rescue VM and attaches the OS Disk of the problem VM to this intermediate working rescue VM to fix the OS Disk issue of the problem VM.
+
+.DESCRIPTION
+    Occasionally Azure IaaS VMs (Microsoft.Compute/virtualMachines) may not start because there is something wrong with the operating system (OS) disk preventing it from booting up correctly.
+    In such cases it is a common practice to recover the problem VM by performing the following steps. These are the steps that are performed by the script.
+	-Stops the problem VM
+	-Take a snapshot of the OS Disk
+	-Create a Temporary Rescue VM
+	-Attach the OS Disk to the Rescue VM
+    -Starts the Rescue VM
+	-RDPs to RescueVM (For Windows)
+
+.PARAMETER VmName
+    This is a mandatory Parameter, Name of the VM that needs to be recovered.
+
+.PARAMETER ResourceGroup
+    This is a mandatory Parameter, Name of the ResourceGroup the VM belongs to.
+
+.PARAMETER SubID
+    This is a mandatory Parameter, SubscriptionID - the VM belongs to.
+
+.PARAMETER prefix
+    Optional Parameter. By default the new Rescue VM and its resources are all created under a ResourceGroup named same as the orginal resourceGroup name with a prefix of 'rescue', however the prefix can be changed to a different value to overide the default 'resuce'
+
+.PARAMETER Sku
+    Optional Parameter. Allows to pass in the SKU of the preferred image of the OS for the Rescue VM
+
+.PARAMETER Offer
+    Optional Parameter. Allows to pass in the Offer of the preferred image of the OS for the Rescue VM
+
+.PARAMETER Publisher
+    Optional Parameter. Allows to pass in the Publisher of the preferred image of the OS for the Rescue VM
+
+.PARAMETER Version
+    Optional Parameter. Allows to pass in the Version of the preferred image of the OS for the Rescue VM
+
+.EXAMPLE
+    .\CreateCRPRescueVM.ps1 -ResourceGroup sujtemp -VmName sujnortheurope -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c ## <==Is an example is with all the mandatory fields
+.EXAMPLE
+    .\CreateCRPRescueVM.ps1 -VmName ubuntu -ResourceGroup rescueportalLin -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c -Publisher RedHat -Offer RHEL -Sku 7.3 -Version 7.3.2017090723 -prefix rescuered <==Examples with optional parametersm in this example it will create the rescue VM with RedHat installed
```

Comment:
>rescueportalLin [](start = 58, length = 15)

Lo-pri: Related to other comment. Might be less confusing to reader if you omit the rescue here